source/fuzz/fuzzer_pass_add_loop_preheaders.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOURCE_FUZZ_FUZZER_PASS_ADD_LOOP_PREHEADERS_H
+#define SOURCE_FUZZ_FUZZER_PASS_ADD_LOOP_PREHEADERS_H
+
+#include "source/fuzz/fuzzer_pass.h"
+
+namespace spvtools {
+namespace fuzz {
+
+// A fuzzer pass that randomly adds loop preheaders to headers that do not have
+// one that satisfies all of the following conditions:
+// - it is the only out-of-loop predecessor of the header
+// - it branches unconditionally to the header
+// - it is not a loop header itself
```

Comment:
I feel that we have made up the term "loop preheader", and a loop preheader must satisfy all of these conditions. Thus, you cannot have a loop preheader that does not satisfy these conditions. Thus, I would say:


A fuzzer pass that randomly adds loop preheaders to loops that do not already have a loop preheader. A loop preheader is simply a predecessor block of the loop header that:
- is the only out-of-loop predecessor of the header
- branches unconditionally to the header
- is not a loop header itself

Reply:
The preheader block definition given [here](https://llvm.org/docs/LoopTerminology.html) only imposes the first two conditions, could this cause any confusion?

Reply:
I changed it to:

A fuzzer pass that randomly adds simple loop preheaders to loops that do not have one. A simple loop preheader is a block that:
- is the only out-of-loop predecessor of the header
- branches unconditionally to the header
- is not a loop header itself